coconut-rs/src/impls/clone.rs

```diff
@@ -0,0 +1,14 @@
+use crate::{Base58, BlindSignRequest, BlindedSignature};
+
+macro_rules! impl_clone {
+    ($struct:ident) => {
+        impl Clone for $struct {
+            fn clone(&self) -> Self {
+                Self::try_from_bs58(self.to_bs58()).unwrap()
```

Comment:
I don't quite understand that, why not just implement clone directly if it's required rather than via the back and forth conversion to base58?

Reply:
Good point, I just wanted something quick and not a lot of code, this gives us blanket implementations for anything that is `impl Base58`. Looking at it now tough it would be cleaner to just use `Bytable` and not bounce through `Base58`